src/cf.data.pre

```diff
@@ -10070,7 +10072,7 @@ TYPE: int
 DEFAULT: 1024
 LOC: Config.ipcache.size
 DOC_START
-	Maximum number of DNS IP cache entries.
+	Maximum number of DNS IP cache entries. A large ipcache size may improve performance because it avoid more address lookups but excessively large values would waste memory. Since the entry only takes very small amounts of memory and the default value is relatively small, the developer may need to set a large size in production.
```

Comment:
Caching data may improve performance. This should be obvious information anyone operating a proxy cache - so no need to mention.

Also performance gain is subjective and dependent on latency of the recursive resolver being used by Squid. So it is possible reduction is better than increase. This is just one of many settings that need network-specific testing before performance tuning, we document performance tuning in the wiki instead of repeating it in every relevant directive.

Reply:
> Caching data may improve performance. This should be obvious information anyone operating a proxy cache - so no need to mention.

I agree with Amos.

FWIW, It would be good to mention the actual size (in bytes) of a typical IP cache entry. _That_ knowledge is specific to Squid implementation and may be relevant for those wanting very large IP caches or those running on a very tight memory budget. Unfortunately, I do not have that information to suggest a specific edit.

P.S. I see problems with parts of the second paragraph in the change request, but I do not think it is important to agree on those issues because the first paragraph (the one quoted above) is enough to reject the proposed Config.ipcache.size documentation update as "common/obvious knowledge".

Reply:
I think documenting the size of a IP cache entry would to good for users to decide the cache size. When I am tunning this configuration,  I felt that tunning this configuration is not that obvious since the performance gain and the memory cost of increasing the ipcache_size is unclear to me. While a large cache would definilty improve the performacne but it requires extra memory.   I search on wiki and find the following two pages: https://wiki.squid-cache.org/Features/CacheManager/IpCache and https://wiki.squid-cache.org/ProgrammingGuide/IpCache, but none of them mentions about how to tune this configuration.  there any document that I miss?

Reply:
That is difficult, because an ipcache entry is dynamic. Size depends on the number of IPs stored and the length of the domain name used to find them.

The only thing we can say for certain is that larger cache entry limit == more memory used. And that is pretty obvious.

FYI: https://wiki.squid-cache.org/SquidFaq/SquidMemory is where memory tuning for performance is documented.

Reply:
> That is difficult, because an ipcache entry is dynamic. Size depends on the number of IPs stored and the length of the domain name used to find them.

Once the sizes of individual components (s4, s6, and c) are known, I see no difficulty in providing a basic formula for the cache size (something along the lines of `n4*s4 + n6*s6 + nd*(sd + c)` but I have not checked the code to come up with the actual formula) so that inquisitive admins can approximate the average entry size for their use cases. And I hope that Cache Manager already reports the actual entries and/or their stats to assist with those calculations.

We can also provide a ballpark entry size estimate for a basic case with one IPv4 and one IPv6 address for some specific average-length domain name.